### PR TITLE
Fixed a few compiler warnings

### PR DESCRIPTION
**Pull request by nfedera**

### CODE REVIEW

**Comment by nfedera** [comment]
These were the warnings:

```
winpr/libwinpr/sysinfo/test/TestGetComputerName.c:247:28:
warning: invalid conversion specifier 'D'
		fprintf(stderr, "%s: (6/%D) GetComputerNameExA failed with error: 0x%08"PRIX32"\n",
                              ~~ ^ ~

--------------------------------------------------------------------------------

winpr/libwinpr/synch/test/TestSynchMultipleThreads.c:124:9:
warning: comparison of unsigned expression < 0 is always false
        if (rc < 0)
            ~~ ^ ~

--------------------------------------------------------------------------------

winpr/libwinpr/synch/test/TestSynchMultipleThreads.c:149:9:
warning: comparison of unsigned expression < 0 is always false
        if (rc < 0)
            ~~ ^ ~

--------------------------------------------------------------------------------

libfreerdp/core/update.c:2159:3:
warning: implicit declaration of function 'update_free_window_icon_info' is invalid in C99

--------------------------------------------------------------------------------

channels/drive/client/drive_file.c:187:17:
warning: assigning to 'WCHAR *' (aka 'unsigned short *') from 'const WCHAR *' (aka 'const unsigned short *') discards qualifiers
        file->fullpath = fullpath;
                       ^ ~~~~~~~~

--------------------------------------------------------------------------------

channels/drive/client/drive_main.c:869:13:
warning: incompatible pointer types passing 'DRIVE_DEVICE *' (aka 'struct _DRIVE_DEVICE *') to parameter of type 'DEVICE *' (aka 'struct _DEVICE *')
        drive_free(drive);
                   ^~~~~

--------------------------------------------------------------------------------

libfreerdp/primitives/test/TestPrimitivesColors.c:105:37:
warning: format specifies type 'unsigned long' but the argument has type 'UINT32' (aka 'unsigned int')
        printf("Results for %lux%lu [%s]", roi.width, roi.height, GetColorFormatName(DstFormat));
                            ~~~            ^~~~~~~~~

--------------------------------------------------------------------------------

libfreerdp/primitives/test/TestPrimitivesColors.c:105:48:
warning: format specifies type 'unsigned long' but the argument has type 'UINT32' (aka 'unsigned int')
        printf("Results for %lux%lu [%s]", roi.width, roi.height, GetColorFormatName(DstFormat));
                                ~~~                   ^~~~~~~~~~

--------------------------------------------------------------------------------

client/common/cmdline.c:2435:43:
warning: comparison of unsigned expression < 0 is always false
                        if ((settings->AutoReconnectMaxRetries < 0) ||
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~

--------------------------------------------------------------------------------

client/X11/xf_gfx.c:298:80:
warning: comparison of unsigned expression < 0 is always false
        if (context->SetSurfaceData(context, surface->gdi.surfaceId, (void*) surface) < 0)
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~

```

**Comment by hardening** [comment]
I wanna fix these from a long time, thanks for doing the job.

File: channels/drive/client/drive_file.c [hunk 1]
```diff
@@ -184,7 +184,7 @@ static BOOL drive_file_remove_dir(const WCHAR* path)
 static void drive_file_set_fullpath(DRIVE_FILE* file, const WCHAR* fullpath)
 {
 	free(file->fullpath);
-	file->fullpath = fullpath;
+	file->fullpath = (WCHAR*) fullpath;
 	file->filename = _wcsrchr(file->fullpath, L'/');
```
Comment (hardening) [inline channels/drive/client/drive_file.c]: just curious, shouldn't we do a copy here or drop the const modifier ?

Comment (nfedera) [inline channels/drive/client/drive_file.c]: @hardening you're right, taking the context into account it is better to drop the const modifier.

File: channels/drive/client/drive_file.c [hunk 2]
```diff
@@ -181,7 +181,7 @@ static BOOL drive_file_remove_dir(const WCHAR* path)
 	return ret;
 }
 
-static void drive_file_set_fullpath(DRIVE_FILE* file, const WCHAR* fullpath)
+static void drive_file_set_fullpath(DRIVE_FILE* file, WCHAR* fullpath)
 {
 	free(file->fullpath);
 	file->fullpath = fullpath;
```
Comment (MartinHaimberger) [inline channels/drive/client/drive_file.c]: i think we should do a copy here, since we free it again in subsequent calls.

Comment (MartinHaimberger) [inline channels/drive/client/drive_file.c]: its not good readable but ok here, forget my last comment.

**Pull request merged**